masq/src/terminal/terminal_interface.rs

```diff
@@ -369,26 +312,45 @@ mod tests {
 
     #[test]
     fn configure_interface_catches_an_error_when_setting_the_prompt() {
-        let subject = interface_configurator(
-            Box::new(producer_of_interface_raw_causing_set_prompt_error),
-            Box::new(result_wrapper_for_in_memory_terminal),
+        let set_prompt_params_arc = Arc::new(Mutex::new(vec![]));
+
+        let result = interface_configurator(
+            Box::new(TerminalWrapper::result_wrapper_for_in_memory_terminal),
+            Box::new(|_name, _terminal| {
+                Ok(InterfaceRawMock::new()
+                    .set_prompt_result(Err(Error::from_raw_os_error(10)))
+                    .set_prompt_params(&set_prompt_params_arc))
```

Comment:
As a convention, I'd like to see xxx_xxx_params before xxx_xxx_result, just because the parameters are supplied before the result is produced, and we do it that way most other places, and it makes it easier to read when you see what you expect to see.